data-avro/src/main/java/com/linkedin/data/avro/DefaultDataToAvroConvertCallback.java

```diff
@@ -137,9 +137,19 @@ protected Object translateField(List<Object> path, Object fieldValue, RecordData
           if (_options.getOptionalDefaultMode() != OptionalDefaultMode.TRANSLATE_TO_NULL &&
               field.getDefault() != null)
           {
-            throw new IllegalArgumentException(
-                message(path,
-                    "cannot translate absent optional field (to have null value) because this field is optional and has a default value"));
+            DataSchema.Type fieldDefaultValueType = field.getType().getType();
+            // punt on other default values for now (too complex to handle)
+            if (fieldDefaultValueType == DataSchema.Type.UNION || fieldDefaultValueType == DataSchema.Type.RECORD
```

Comment:
In #133, it already ruled out UNION case? Also for TYPEREF, we should look at dereferencedType?